lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -151,8 +151,37 @@ def which(cmd):
     return None
 
 
-class MPI(object):
+class RunFileLoopMPI(RunFile, RunLoop):
+    """
+    RunFile specific for MPI because intel doesn't feel like fixing their shit
+    """
+    def __init__(self, cmd, **kwargs):
+        self.options = kwargs.pop('options', None)
+        self.counter = 1
+        super(RunFileLoopMPI, self).__init__(cmd, **kwargs)
+        self.readsize = -1
+        self.seen_output = False
+
+    def _loop_process_output(self, output):
+        """
+        check if process is generating any output at all; if not, warn the user after a set amount of time
+        """
+        del output
+        if not self.seen_output:
+            self.readfile = open(self.filename, 'r')
+            with open(self.options.output, 'r') as fin:
+                if len(fin.read())>0:
+                    self.seen_output = True
+
+            time_passed = self._loop_count * self.LOOP_TIMEOUT_MAIN
+            if time_passed > self.options.timeout * self.counter and not self.seen_output:
+                print '\n'.join(["WARNING: mympirun has been running for %s seconds without any output." % time_passed,
+                "This means your program may be hanging. Please check if this is normal.",
+                "If not, check your SSH key permissions? (private key should be protected)"])
```

Comment:
compose the message first, and print it to `stderr`
I also wouldn't mention the SSH key thing, since it may not be entirely clear to users *what* to check exactly, and there are other possible reasons for a hang

```python
warning_msg = '\n'.join([
    ""WARNING: mympirun has been running for %s seconds without seeing any output." % time_passed,
    "This may mean that your program is hanging, please check and make sure that is not the case!",
    '',
    "If this warning is printed too soon and the program is doing useful work without producing any output,",
    "you can increase the timeout threshold via --timeout (current setting: %s seconds)" % self.timeout,
])
sys.stderr.write(warning_msg + '\n')
```